programs/governance-rewards/src/instructions/create_distribution.rs

```diff
@@ -0,0 +1,89 @@
+use anchor_lang::prelude::*;
+use std::mem::size_of;
+
+use crate::{
+    error::GovernanceRewardsError,
+    state::{distribution::Distribution, distribution_option::DistributionOptions},
+};
+
+/**
+ * Instruction to create a Distribution.
+ *
+ * The caller must provide a registration_cutoff, which specifies the timestamp at
+ * which registration will end.
+ */
+#[derive(Accounts)]
+pub struct CreateDistribution<'info> {
+    /**
+     * Address of the distribution to be created.
+     */
+    #[account(
+        init,
+        payer = payer,
+        space = 8 + size_of::<Distribution>()
+    )]
+    pub distribution: Box<Account<'info, Distribution>>,
+
+    /**
+     * Account to own any provided token accounts.
+     */
+    /// CHECK: Not read
+    #[account(seeds = [b"payout authority".as_ref(), distribution.key().as_ref()], bump)]
+    pub payout_authority: AccountInfo<'info>,
```

Comment:
It's an optional change, but API-wise, the `payout_authority`, `realm` and `voter_weight_program` could just be Pubkey arguments to the instruction, instead of being used as accounts. 

This would also apply to "Not read" accounts in other instructions.

Reply:
Do you think it's better to have them as arguments?

Reply:
Techincally I prefer it - there's no need to request these accounts to be loaded by the runtime. But client side it leads to worse api for the PDA account because auto-derivation no longer works. So overall I'm ok for it to stay as-is.

Reply:
In future I'll maybe switch.